client-java/src/unit-tests/java/uk/gov/gchq/palisade/client/FullTest.java

```diff
@@ -78,11 +78,11 @@ void testWithDownloadOutsideStream() throws Exception {
         var download = session.fetch(resource);
         assertThat(download).isNotNull();
 
-        var actual = download.getInputStream();
-        var expected = Thread.currentThread().getContextClassLoader().getResourceAsStream("resources/pi0.txt");
-
-        assertThat(actual).hasSameContentAs(expected);
-
+        try (var actual = download.getInputStream();
+             var expected = Thread.currentThread().getContextClassLoader().getResourceAsStream("resources/pi0.txt");
+        ) {
+            assertThat(actual).hasSameContentAs(expected);
+        }
```

Comment:
If you declare the exception at the top with the throws Exception, what's the benefit to surrounding this in a try?

Reply:
`InputStream` implements `AutoCloseable`, so `actual` and `expected` will be closed if not already done so by assertj. Stops resource leaks.

Reply:
Sorted